### ISSUE TITLE
Migration error after adding #2 compatibility_test migration

### ISSUE THREAD

**Issue by chompomonim** [ManoSeimas/manoseimas.lt#159]
$ make migrate
...

```
Applying scrapy.0001_initial... OK
  Applying compatibility_test.0001_initial... OK
  Applying compatibility_test.0002_auto_20160531_1913...Traceback (most recent call last):
  File "bin/django", line 98, in <module>
    sys.exit(djangorecipe.binscripts.manage('manoseimas.settings.vagrant'))
  File "/home/vagrant/manoseimas/eggs/djangorecipe-2.1.2-py2.7.egg/djangorecipe/binscripts.py", line 9, in manage
    management.execute_from_command_line(sys.argv)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/core/management/__init__.py", line 351, in execute_from_command_line
    utility.execute()
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/core/management/__init__.py", line 343, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/core/management/commands/migrate.py", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/migrations/executor.py", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/migrations/executor.py", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/migrations/migration.py", line 115, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/migrations/operations/fields.py", line 121, in database_forwards
    schema_editor.remove_field(from_model, from_model._meta.get_field(self.name))
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/backends/base/schema.py", line 440, in remove_field
    self.execute(sql)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/backends/base/schema.py", line 111, in execute
    cursor.execute(sql, params)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/utils.py", line 97, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/manoseimas/eggs/Django-1.8.5-py2.7.egg/django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "/home/vagrant/manoseimas/eggs/MySQL_python-1.2.5-py2.7-linux-x86_64.egg/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/vagrant/manoseimas/eggs/MySQL_python-1.2.5-py2.7-linux-x86_64.egg/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
django.db.utils.OperationalError: (1025, "Error on rename of './manoseimas/#sql-48b_99' to './manoseimas/compatibility_test_politicaltopic' (errno: 150)")
make: *** [migrate] Error 1
```

**Comment by sirex** [comment]
@chompomonim I have tested migrations on empty database, on imported production database, on Travis and running migrations in tests. Can't reproduce it. So I guess this is error is specific with your database state. How you came to this state, I don't know, and note sure if it worth trying to figure this out.

I would recommend to reset you database with `make reset_mysql` and then run migrations again with `make migrate`.

**Comment by frgtn** [comment]
I've also seen this error during deployment into an empty Vagrant box.

**Comment by sirex** [comment]
@frgtn not sure what exactly causes this, but @chompomonim found, that removing these lines:

https://github.com/ManoSeimas/manoseimas.lt/blob/new_test/manoseimas/settings/vagrant.py#L13-L14

```python
'init_command': (
    'SET storage_engine=INNODB; '
    'SET GLOBAL sql_mode=STRICT_ALL_TABLES;'
)
```

Issue disappears.

**Comment by chompomonim** [comment]
Yes, try to delete `DATABASES = {...}` from manoseimas/settings/vagrant.py and it should work.

Actually not sure why we needed `SET GLOBAL sql_mode=STRICT_ALL_TABLES;` in the past. It's only one difference with base.py settings, os I just use base.py settings for DATABASES and it works.

**Comment by frgtn** [comment]
`SET GLOBAL sql_mode=STRICT_ALL_TABLES;` was added to make sure errors are raised when legacy MySQL behaviour would try to silently truncate/drop data being inserted, see http://dev.mysql.com/doc/refman/5.6/en/sql-mode.html#sql-mode-strict

I would be really careful removing `sql_mode=STRICT_ALL_TABLES` as it would hide data coercion issues that might be difficult to debug and fix once the code gets to prod.

As for `SET storage_engine=INNODB;` I *think* it was only ever needed to make sure database created by Django for tests was `InnoDB` and not `MyISAM` but I might be wrong. The dev (not the test) database should already be `InnoDB`.

Just shooting in the dark, but @chompomonim can you check what database type is your dev database?

**Comment by chompomonim** [comment]
Why we're not adding it to base config then?

**Comment by frgtn** [comment]
`¯\_(ツ)_/¯`

**Comment by frgtn** [comment]
I guess `STRICT_ALL_TABLES` was never turned on for non-vagrant dev. I guess it's reasonable to disable it as backtracking and making sure all the past migrations work with `STRICT_ALL_TABLES` will a pain / cost a lot of time.

**Comment by menesis** [comment]
InnoDB is the default engine since MySQL 5.5 that shipped with Ubuntu 12.04, so no need to set that explicitly.
